### PR TITLE
Ming/demo

### PR DESCRIPTION
**Pull request by mingsorawit**
### Problems

- created demo page which is will have 3 pages
- created chat page, payment page, receipt page

28/7
- format files 

29/7
- add animation for transitioning between pages

30/7
- add asset in  demo page1,2
- add animation for slide to pay button
- update copywriting in every demo page
- create Demo component with css
- replace phone screen picture with a Demo component in index.tsx

2/8
- add beating animation for clickable area
- resolve issues from feedback( cursor, responsive version)
- add 'try demo tag'
- change button ui in last page

3/8
- make checkout link clickable, responsive version for ipad portrait

### How would the reviewers verify the solutions

add animation

https://user-images.githubusercontent.com/70103668/127466215-0e59be96-b597-4393-9399-12bea55d8328.mp4

### Preflight checklist

- [ ] **Manual testing** has been performed according to the acceptance criteria.
- [ ] _(heavy style change)_ The solution has been tested on the major browser (Chrome, Firefox, Safari, Microsoft Edge).

### CODE REVIEW

File: pages/demo.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import styles from '../styles/demo.module.css'
+import ChatPage from '../components/demo_pages/ChatPage'
+import PaymentPage from '../components/demo_pages/PaymentPage'
+import ReceiptPage from '../components/demo_pages/ReceiptPage'
+
+export default function Demo() {
+  const [pageNumber, setPageNumber] = useState(1)
+
+  const nextPage = () => {
+    setPageNumber(pageNumber + 1)
```
Comment (oukunanan) [inline pages/demo.tsx:12]: SetX(value =》 value+1)

File: pages/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  const firstPage = () => {
```
Comment (oukunanan) [inline pages/demo.tsx:15]: Better name like `backToFirstPage`

File: components/demo_pages/ReceiptPage.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            <div className={styles.child_key}>Merchant</div>
+            <div>XYZ Furniture</div>
+          </div>
+          <div className={styles.info_child}>
+            <div className={styles.child_key}>Payment</div>
+            <div>HSBC*1234</div>
+          </div>
+          <div className={styles.info_child}>
+            <div className={styles.child_key}>Amount</div>
+            <div>8,400 THB</div>
```
Comment (oukunanan) [inline components/demo_pages/ReceiptPage.tsx:48]: Span

File: components/demo_pages/PaymentPage.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import styles from './PaymentPage.module.css'
+import React from 'react'
+import Image from 'next/image'
+import cardIcon from '../../public/card-icon.png'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faPen, faChevronRight } from '@fortawesome/free-solid-svg-icons'
+
+type Props = {
+  onClick: () => void
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.tsx:9]: Better props name

File: components/demo_pages/PaymentPage.module.css [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  align-items: center;
+  display: flex;
+}
+
+.row_bottom {
+  margin-top: 150px;
+  border-top: solid 2px black;
+  padding: 8px 0px 12px 0px;
+  display: flex;
+  flex-direction: row;
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.module.css:102]: Remove

File: components/demo_pages/PaymentPage.module.css [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  font-size: 16px;
+  border-radius: 10px;
+  box-shadow: 0 5px 20px 0 #dbe2ff;
+  line-height: 1.3;
+  gap: 4px;
+}
+
+.info_box_payment {
+  display: flex;
+  flex-direction: row;
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.module.css:77]: Remove

File: components/demo_pages/ChatPage.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  faBatteryFull,
+  faSignal,
+  faEllipsisH,
+  faPlusCircle,
+  faCamera,
+  faPaperPlane,
+} from '@fortawesome/free-solid-svg-icons'
+
+type Props = {
+  onClick: () => void
```
Comment (oukunanan) [inline components/demo_pages/ChatPage.tsx:18]: Better props name, what is `onClick` do?

File: components/demo_pages/PaymentPage.module.css [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+  gap: 4px;
+}
+
+.wrapper_blue {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 4px;
+  color: #2c46b5;
+}
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.module.css:28]: A lot of redundant properties, I think we can do something like this
```suggestion
.wrapper {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 4px;
}

.wrapper_blue {
  color: #2c46b5;
}
```

File: components/demo_pages/PaymentPage.module.css [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+.header {
+  display: flex;
+  flex-direction: row;
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.module.css:32]: Remove

File: components/demo_pages/PaymentPage.module.css [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+  font-family: 'Assistant';
+  font-size: 12px;
+  height: 16px;
+  margin-top: 20px;
+  color: #535353;
+}
+
+.id_and_ref {
+  display: flex;
+  flex-direction: row;
```
Comment (oukunanan) [inline components/demo_pages/PaymentPage.module.css:44]: Remove

File: components/demo_pages/ChatPage.module.css [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+  align-items: center;
+}
+
+.text_input {
+  width: 200px;
+  height: 40px;
+  border-radius: 20px;
+  color: lightgray;
+  background-color: white;
+  padding: 10px 0px 0px 18px;
```
Comment (oukunanan) [inline components/demo_pages/ChatPage.module.css:131]: This cannot guarantee text vertical-align, fine a better approaches

File: components/demo_pages/ChatPage.module.css [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+.message_left {
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: flex-end;
+  padding-top: 12px;
+  padding-left: 44px;
+  position: relative;
+}
```
[the remark this answers is not in the document]
Comment (mingsorawit) [reply components/demo_pages/ChatPage.module.css:80]: Created a new class called 'message_row', that contains the same properties.
Different properties are in 'row_align_left' and 'row_align_right'.

File: components/demo_pages/ChatPage.module.css [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+  padding: 14px 20px 14px 32px;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  background-color: #5b5e6e;
+  color: white;
+}
+
+.wrapper {
```
[the remark this answers is not in the document]
Comment (mingsorawit) [reply components/demo_pages/ChatPage.module.css:18]: a wrapper for any component that have to stick together(with small gap), such as the icons in top right corner.

File: components/demo_pages/ChatPage.module.css [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+  padding: 12px 15px;
+  line-height: 1.4;
+}
+
+.message_box_first_left {
+  background-color: #f4f6ff;
+  border-radius: 15px 15px 15px 0px;
+  padding: 12px 15px;
+  line-height: 1.4;
+}
```
[the remark this answers is not in the document]
Comment (mingsorawit) [reply components/demo_pages/ChatPage.module.css:107]: For these 3 classes, the same properties are in 'message_box', and different border-radius property is in 'box_first_right' and 'box_first_left'.

[the remark this answers is not in the document]
Comment (oukunanan) [reply components/demo_pages/ChatPage.module.css]: Can we do with this [approches](https://github.com/Beam-Data-Company/beam-early-access/pull/11#discussion_r677459699) ?

File: styles/demo.module.css [hunk 15]
```diff
@@ -2,5 +2,15 @@
...
+  height: 770px;
+  position: relative;
+  overflow: hidden;
+}
+
+.temp {
+  position: absolute;
+  top: 0%;
+  left: 50%;
+  transform: translate(-50%, 0%);
```
Comment (oukunanan) [inline styles/demo.module.css:14]: Checking

File: components/demo_pages/ReceiptPage.tsx [hunk 16]
```diff
@@ -6,7 +6,7 @@ import confetti from '../../public/confetti.png'
 import checkIcon from '../../public/check-icon.png'
 
 type Props = {
-  onClick: () => void
+  buttonHandleClick: () => void
```
Comment (oukunanan) [inline components/demo_pages/ReceiptPage.tsx:9]: Better props name

File: pages/demo.tsx [hunk 17]
```diff
@@ -8,46 +8,80 @@ import ReceiptPage from '../components/demo_pages/ReceiptPage'
...
 
-  const backToFirstPage = () => {
-    setPageNumber(1)
+  type Props = {
+    pageNumber: number
   }
 
-  const displayPage = () => {
-    if (pageNumber === 1) {
+  const Page1 = (props: Props) => {
```
Comment (oukunanan) [inline pages/demo.tsx:19]:
```ts
const Page1 = ({ pageNumber }: { pageNumber: number })
```

File: components/demo_pages/ChatPage.tsx (hunk 7 above)
Comment (mingsorawit) [reply components/demo_pages/ChatPage.tsx]: changed to 'goToPaymentPage'

File: components/demo_pages/ChatPage.module.css (hunk 11 above)
Comment (mingsorawit) [reply components/demo_pages/ChatPage.module.css]: changed to display flex

File: components/demo_pages/ReceiptPage.tsx (hunk 16 above)
Comment (mingsorawit) [reply components/demo_pages/ReceiptPage.tsx:9]: changed to 'goToChatPage'

File: pages/demo.tsx (hunk 17 above)
Comment (mingsorawit) [reply pages/demo.tsx:19]: done

**Review by oukunanan — APPROVED** [review]
Approved in advance

**Pull request merged**